src/data/polynomial/algebra_map.lean

```diff
@@ -97,17 +97,15 @@ end comm_semiring
 section aeval
 variables [comm_semiring R] {p : polynomial R}
 
--- TODO this could be generalized: there's no need for `A` to be commutative,
--- we just need that `x` is central.
-variables [comm_semiring A] [algebra R A]
-variables {B : Type*} [comm_semiring B] [algebra R B]
+variables [semiring A] [algebra R A]
+variables {B : Type*} [semiring B] [algebra R B]
 variables (x : A)
 
 /-- Given a valuation `x` of the variable in an `R`-algebra `A`, `aeval R A x` is
 the unique `R`-algebra homomorphism from `R[X]` to `A` sending `X` to `x`. -/
 def aeval : polynomial R →ₐ[R] A :=
 { commutes' := λ r, eval₂_C _ _,
-  ..eval₂_ring_hom (algebra_map R A) x }
+  ..eval₂_ring_hom' (algebra_map R A) (λ a b, (algebra.commutes a b).symm) x }
```

Comment:
Would this just be `algebra.commutes` instead of `(λ a b, (algebra.commutes a b).symm)` if you flip the equality in the definition? It seems worth it to optimize for that.

Reply:
I think I've fixed this, but we'll see in a few minutes if I managed to fix the merge conflict I created in the process.